dvc/tree/repo.py

```diff
@@ -336,13 +342,17 @@ def walk(
                 onerror(NotADirectoryError(top))
             return
 
+        ignore_subrepos = not self._traverse_subrepos
+        if follow_subrepos is not None:
+            ignore_subrepos = not follow_subrepos
+
         tree, dvc_tree = self._get_tree_pair(top)
         repo_exists = tree.exists(top)
         repo_walk = tree.walk(
             top,
             topdown=topdown,
             onerror=onerror,
-            ignore_subrepos=not self._traverse_subrepos,
+            ignore_subrepos=ignore_subrepos,
```

Comment:
We were confusing seeing and following subrepos before, but in the future there should be two options: `ignore_subrepos` and `follow_subrepos`.